### DISCUSSION TITLE
POST requests are blocked when creating many documents

### DISCUSSION THREAD

**Discussion by MedLabs** [pocketbase/pocketbase#2407] [Q&A] [answered]
Hello,
I have a little custom syncing function that takes an array of documents and uses a for loop to run pb.collection("collection").create(doc)
when I run this function it adds only two docs then I get a bunch of Blocked POST requests.

what am I doing wrong ? is there any bulkAdd method ? I use js-sdk

**Comment by tigawanna** [comment]
Are you using promise.all?

**Comment by ganigeorgiev** [comment, marked as answer]
> it adds only two docs then I get a bunch of Blocked POST requests.

If you are sending the requests concurrently, then this is most likely due to [autocancellation](https://github.com/pocketbase/js-sdk/tree/master#auto-cancellation). You can disable it globally using `pb.autoCancellation(false)`. For example:

```js
const pb = new PocketBase("http://127.0.0.1:8090");

pb.autoCancellation(false);

const docs = [
    ...
];

const promises = [];

for (let doc of docs) {
    promises.push(pb.collection("example").create(doc));
}

await Promise.all(promises);
```

> is there any bulkAdd method ?

No, at the moment there are no bulk actions.
